core/scheduler/service/job_run_service.go

```diff
@@ -146,6 +149,27 @@ func (s *JobRunService) JobRunInput(ctx context.Context, projectName tenant.Proj
 	return s.compiler.Compile(ctx, details, config, executedAt)
 }
 
+func (s *JobRunService) GetJobRunsByFilter(ctx context.Context, projectName tenant.ProjectName, jobName scheduler.JobName, filters ...filter.FilterOpt) ([]*scheduler.JobRun, error) {
```

Comment:
please also add test for this function as well

Reply:
sure